### ISSUE TITLE
Reuse code among YarpPlugins

### ISSUE THREAD

**Issue by jgvictores** [roboticslab-uc3m/openrave-yarp-plugins#50]
WIP with `YarpOpenraveGrabber` and `YarpOpenraveRGBDSensor`.

**Comment by jgvictores** [comment]
WIP at branch `issue-50-YarpOpenrave-reuse`, current commit 0d70c24.

**Comment by jgvictores** [comment]
Blocking #49.

**Comment by jgvictores** [comment]
Code reuse of `penv`, `probot`, and `robotName` for YarpPlugins implemented and merged into `develop` at 140c342.

Must decide on sensors and manipulators.

**Comment by jgvictores** [comment]
Blocked by #51, which will define more on how sensor plugins are implemented.

**Comment by jgvictores** [comment]
Now that #51 has been closed, some basic reflections.

Had an idea of making an additional abstraction layers between `OpenraveYarpBase` and rest of `YarpPlugins` of this repository, something like `OpenraveYarpSensorBase` and `OpenraveYarpManipulatorBase`. However, while it was possible to find common elements for `OpenraveYarpBase` (an environment, a robot), `YarpOpenraveRGBDSensor` has proved that a general `OpenraveYarpSensorBase` with a standarized `sensorIndex` is not possible, as the same plugin may use several sensors (e.g. in sensors such as RGBD, Stereo,...). While this is solvable via a plural `sensorIndices` parameter, I wasn't looking only for shared parameters, but also very generic code, which does not seem to be the case for now. Additionally, even the assumption that plugins can be categorized in two big groups: `manipulators` (for controlboards), and `sensors`; is yet to be proved. Food for thought.

**Comment by jgvictores** [comment]
Changing label to `idea` and removing `blocking` status (see comments on #49 which was blocked).

**Comment by jgvictores** [comment]
Closing issue, as [libraries/YarpOpenraveBase](https://github.com/roboticslab-uc3m/openrave-yarp-plugins/tree/4f38b305f759f6c7335bcf02eb74cbbe72190513/libraries/YarpOpenraveBase) has been working for a while.
The setup is that YARP plugins are individual devices of a robot, so up to robot level is shared. Lower level `sensorIndex` are trated below ([example](https://github.com/roboticslab-uc3m/openrave-yarp-plugins/blob/4f38b305f759f6c7335bcf02eb74cbbe72190513/libraries/YarpPlugins/YarpOpenraveGrabber/DeviceDriverImpl.cpp#L23)) and collections such as `allManipulators` may and are treated above ([example](https://github.com/roboticslab-uc3m/openrave-yarp-plugins/blob/4f38b305f759f6c7335bcf02eb74cbbe72190513/libraries/OpenravePlugins/OpenraveYarpPluginLoader/OpenraveYarpPluginLoader.cpp#L220)).
